### PR TITLE
feat: add optional nginx config for knowledge base url

### PR DESCRIPTION
**Pull request by jasjukaitis**
Signed-off-by: Raphael Jasjukaitis <hello@raphael-jasjukaitis.de>

#### What this PR does / why we need it:
Zammad has the feature to add a custom url for the knowledge base. If enabled, `knowledgeBaseUrl` can be set so nginx will be configured accordingly.

#### Which issue this PR fixes
-

#### Special notes for your reviewer:
-

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: zammad/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: zammad
-version: 6.0.4
+version: 6.0.5
```
Comment (monotek) [inline zammad/Chart.yaml:3]: New feature should raise minor version...

```suggestion
version: 6.1.0
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by MrGeneration** [comment]
Sorry for bothering but this seems to be half backed.
Zammads knowledge base allows either a different location than `/help` or your own subdomain.
This approach seems to support the subdomain approach only.

**Comment by monotek** [comment]
How is it solved in general?
Could'nt find any docs about it?

**Comment by MrGeneration** [comment]
> How is it solved in general?

Upon configuration Zammad provides you with the required configurations for nginx and apache.
So I copied them out of the application for your context:

different directory (`/myhelp`)
```
# Add following lines to "server" directive
rewrite ^/myhelp(.*)$ /help$1 last;

# Add following line to "Location /" directive, before other proxy_set_header
proxy_set_header X-ORIGINAL-URL $request_uri;
```

custom subdomain (`help.example.com`)
```
# Add following lines to "server" directive
if ($host = help.example.com ) {
  rewrite ^/(api|assets)/(.*)$ /$1/$2 last;
  rewrite ^(.*)$ /help$1 last;
}

# Add following line to "Location /" directive, before other proxy_set_header
proxy_set_header X-ORIGINAL-URL $request_uri;
```



> Could'nt find any docs about it?

https://admin-docs.zammad.org/en/latest/manage/knowledge-base.html#custom-url

**Comment by jasjukaitis** [comment]
Got it, just tried different combinations for snippets. Also tried to use the elegant way with `{{- $url := urlParse .Values.zammadConfig.nginx.knowledgeBaseUrl }}` but unfortunately `urlParse` does not detect `foo.example.com` as a host, it's the path for it. Bummer…

**Review by monotek — DISMISSED** [review]

**Pull request merged**